storage/cache/log_subtree_cache.go

```diff
@@ -35,6 +35,49 @@ func LogPopulateFunc(hasher hashers.LogHasher) storage.PopulateSubtreeFunc {
 	return populateLogSubtreeNodes(hasher)
 }
 
+func suffixKey(d, i int) string {
+	sfx := &storage.Suffix{Bits: byte(d), Path: []byte{byte(i << uint(8-d))}}
+	maskLowBits := (sfx.Bits-1)%8 + 1
+	sfx.Path[0] &= ((0x01 << maskLowBits) - 1) << uint(8-maskLowBits)
+	return sfx.String()
+}
+
+func populatePerfectLogSubtree(hasher hashers.LogHasher, st *storagepb.SubtreeProto) error {
+	if got, want := len(st.Leaves), 256; got != want {
+		return fmt.Errorf("subtree is not perfect, tree has %d leaves, expected %d", got, want)
+	}
+	var l, r []byte
+	var ok bool
+	for d := 7; d >= 0; d-- {
+		for i := 0; i < 1<<uint(d); i++ {
+			leftChild := i * 2
+			childDepth := d + 1
```

Comment:
When I saw that I thought it was going outside the subtree for d=7. Comment about leaves maybe?